src/frontend/src/components/ForgotPassword/index.tsx

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { AppState } from '../../redux/store';
+import { logIn } from '../../redux/system/actions';
+import { SystemState } from '../../redux/system/types';
+import Logo from '../Logo';
+import { Link } from 'react-router-dom';
+
+interface Props {
+  system: SystemState;
+  logIn: typeof logIn;
+}
+interface State {
+  email: string;
+  emailHasInput: boolean;
+  emailIsValid: null | boolean;
+  missingInputs: null | boolean;
+}
+
+class ForgotPassword extends React.Component<Props, State> {
+  state: State = {
+    email: '',
+    emailHasInput: false,
+    emailIsValid: null, // email validation to be added
+    missingInputs: null
+  };
+
+  handleChange = (e: React.BaseSyntheticEvent) => {
+    // See https://github.com/DefinitelyTyped/DefinitelyTyped/issues/26635 for why we're
+    // using the "any" type.
+    this.setState<any>({
+      [e.target.id]: e.target.value
+    });
+  };
+
+  handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+    this.validateForm();
+  };
+
+  emailCheck = (email: string) => {
+    return /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(email);
+  };
+
+  validateForm = () => {
+    this.setState({
+      emailHasInput: this.state.email.trim().length === 0
```

Comment:
Should this be `!==`?

Reply:
Ah, this kept tripping me up. The logic is correct but I've renamed the property "ariaInvalid" as I'm using it to set that property on the input field.